interceptor.go

```diff
@@ -35,17 +36,15 @@ import (
 // OpenTelemetry metrics and tracing to connect handlers and clients.
 type Interceptor struct {
 	config            config
-	clientOnce        sync.Once
 	clientInstruments instruments
-	serverOnce        sync.Once
 	serverInstruments instruments
 }
 
 var _ connect.Interceptor = &Interceptor{}
 
 // NewInterceptor constructs and returns an Interceptor which implements [connect.Interceptor]
 // that adds OpenTelemetry metrics and tracing to Connect handlers and clients.
-func NewInterceptor(options ...Option) *Interceptor {
+func NewInterceptor(options ...Option) (*Interceptor, error) {
```

Comment:
This needs new doc comments about the conditions in which it can return an error.

Are instruments only used for metrics, and not needed for tracing? If so, can metrics be disabled via options/config (like an interceptor that _only_ does tracing)? If that is possible, we'd only ever want to return an error when metrics are actually enabled.

BTW, I think we actually have another option here, other than breaking the API: we could add a new option to capture the error, like `OnInstrumentError(*error)` that stores an error at the given address when instruments could not be initialized. Using an option would put the onus on the caller as to whether to ignore the returned interceptor or not. They may choose that it's acceptable to ignore the error and have an interceptor that only does tracing. Whereas without an option, we're always returning a nil interceptor on error (even if there may be valuable in a tracing-only interceptor).

It actually feels like tracing and metrics want to be configured separately. Maybe something to think about in a new API.

Reply:
Updated the doc to help clarify the error cases and how to set options to customize tracing or metric providers. Will think about how the API could be improved to provide better fallbacks on error. For now you could provide the same fallback by inspecting the error and then initializing without metrics enabled. Something like:
```go
interceptor, err := otelconnect.NewIntercepetor(opts...)
if err != nil {
    otel.Handle(err) // log error and debug metrics implementation later
    interceptor, err = otelconnect.NewInterceptor(append(opts, otelconnect.WithoutMetrics()))
    if err != nil {
        return err
    }
}
```

Reply:
The code unconditionally configures the instruments, even if metrics are disabled. I guess that means that the no-op meter never returns any errors?

Reply:
Yep, don't need to conditionally apply it as noop won't error. All instrument calls will still be invoked by use the noop meter.